### PR TITLE
fix: Outdated design for Rest API Editor

### LINKED ISSUE #13330: [Type]-[12000]:Outdated Design for Rest API Editor
### Is there an existing issue for this?

- [X] I have searched the existing issues

It will fix following issues:
- [Inconsistent datasource dropdown](https://www.notion.so/appsmith/Usability-UI-issues-3ed292f1e7924526b2a3bd56f0a50b5c?p=fb227188f2fc49c0a74dbd899a9a0948)
- [bgColor mismatch for method dropdown](https://www.notion.so/appsmith/Usability-UI-issues-3ed292f1e7924526b2a3bd56f0a50b5c?p=f9cec4b3fe424839b14b94636bb73861)
- [Method dropdown is styled differently](https://www.notion.so/appsmith/Usability-UI-issues-3ed292f1e7924526b2a3bd56f0a50b5c?p=e841eae7c96643679500ac6896cb8832)
- [Slash icon alignment in Key Value component](https://www.notion.so/appsmith/Usability-UI-issues-3ed292f1e7924526b2a3bd56f0a50b5c?p=d7b32f8f3a3447a98741560988e32080)

### SubTasks

- Design and implement new REST API Editor as per new design system

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
## Description

> Changed background color and fixed the padding along with margin in quick access tip
>Fixed selected and hover color in http method dropdown
>Proper alignment of slash icon in Input editor.

Fixes #13330

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

**Review by Irongade — DISMISSED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]

**Comment by yatinappsmith** [comment]
Testplan link 
- [ ]  [1731 [TestCase] Design tests for REST api(Bugs) ](https://github.com/appsmithorg/TestSmith/issues/1731)

**Review by Irongade — DISMISSED** [review]

**Comment by btsgh** [comment]
Tested for the changes mentioned in the issue - https://github.com/appsmithorg/appsmith/issues/13330
All the changes seem to have been incorporated.

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/ApiTests/API_Styles_spec.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  const backgroundColorwhite = "rgb(255, 255, 255)";
+  const fontColorGray800 = "rgb(57, 57, 57)";
+
+  before(() => {
+    //Create test api
+    cy.NavigateToAPI_Panel();
+    cy.CreateAPI("test_styles");
+  });
+
+  it("Quick access command background color", function() {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/ApiTests/API_Styles_spec.js:19]: Give number to "it" functions - which will make the debugging easier incase of CI failures. ex: `it("1. Quick access command background color", function()`

**Review by Aishwarya-U-R — CHANGES_REQUESTED** [review]

**Review by Aishwarya-U-R — APPROVED** [review]
Approving cypress tests!

**Pull request merged**